rpc/network/network.go

```diff
@@ -83,16 +84,27 @@ func (nm *Manager) Init(networks []params.Network) error {
 		return nil
 	}
 
+	var errors string
 	currentNetworks, _ := nm.Get(false)
-	if len(currentNetworks) > 0 {
```

Comment:
This condition here will, in all cases but when you're running the app for the first time, be true. So newly added networks won't be added to the db, `networks` table ever. So in the for loop below we're checking if the network with `ChainID` exists in the table and if not, we're adding it.

But I think that we should overwrite entries always since `rpc_url` contains infura key which can be changed depends we're running the app locally or one built on CI.

Reply:
> This condition here will, in all cases but when you're running the app for the first time, be true. So newly added networks won't be added to the db, networks table ever. So in the for loop below we're checking if the network with ChainID exists in the table and if not, we're adding it.

Yes agree with that

> But I think that we should overwrite entries always since rpc_url contains infura key which can be changed depends we're running the app locally or one built on CI. Do you agree?

I agree with this statement, but we also need to take in consideration that user can update the network. We may be need a flag `changedByUser` and if true, don t do the update